Nonprofit-Exchange-Hub/web-app
# feat: categories FE/BE hookup

## Dev Summary
[Mocks](https://www.figma.com/file/KE2Sw6ob07agN25RmeNuFg/GiveCycle-wireframes?node-id=1936%3A3444)

Retrieves categories from the backend and populates the field on the goods request form (closes #115)

## Test Plan

1. Load both servers
2. Create some categories, e.g. in Postman, import from [this link](https://www.getpostman.com/collections/368919bc2613efca14fb) and run the post requests (1-5). This will create four categories with `applies_to_assets` set to true, and one with `applies_to_assets` set to false

<img width="378" alt="Screenshot 2022-03-01 at 03 26 13" src="https://user-images.githubusercontent.com/40244233/156099210-8ae5f1dd-73d3-4a50-b9c6-82d29324de14.png">

3. Navigate to the [Share a Need: Goods form](http://localhost:3000/need/goods)
4. Click on the categories input

**Expected:** four categories show, as below

<img width="687" alt="Screenshot 2022-03-01 at 03 27 52" src="https://user-images.githubusercontent.com/40244233/156099374-72e389cc-c2b3-46d9-bc51-f406fb9985a3.png">

## Comments

Had a couple of user-related warnings preventing pushing (because of the Husky hooks). Commented out one of these where I don't think it's yet being used and is incomplete anyway (reset password), but for the other (in client/src/providers/UserProvider.tsx) I just disabled ESLint for that line. Had a play around to try to fix it, but couldn't see a suitable solution from some quick research. Most suggest moving definition of the function called in the `useEffect` block inside the block, but it's used in the return statement outside that block
